### PR TITLE
refactor: change twilio error handling to switch

### PR DESCRIPTION
**Pull request by vindard**
## Description

Change to switch and add new cases to handle these unknown errors: https://ui.honeycomb.io/galoy/datasets/galoy-bbw/result/gu6iWKxazW7?hideMarkers

### CODE REVIEW

File: src/services/twilio.ts [hunk 1]
```diff
@@ -33,27 +33,26 @@ export const TwilioClient = (): IPhoneProviderService => {
...
 
-      if (err.message.includes("timeout of") && err.message.includes("exceeded")) {
-        return new PhoneProviderConnectionError(err)
-      }
+        case match(KnownTwilioErrorMessages.BadPhoneProviderConnection):
+          return new PhoneProviderConnectionError(err)
 
-      return new UnknownPhoneProviderServiceError(err)
+        default:
+          return new UnknownPhoneProviderServiceError(err)
```
Comment (dolcalmi) [inline src/services/twilio.ts:54]: please use `err.message || err`

File: src/services/twilio.ts [hunk 2]
```diff
@@ -77,6 +76,10 @@ export const TwilioClient = (): IPhoneProviderService => {
         return new InvalidPhoneNumberPhoneProviderError(err.message || err)
       }
 
+      if (err.message.includes("timeout of") && err.message.includes("exceeded")) {
+        return new PhoneProviderConnectionError(err)
```
Comment (dolcalmi) [inline src/services/twilio.ts:80]: same here

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: src/services/twilio.ts (hunk 2 above)
Comment (vindard) [reply src/services/twilio.ts]: Done: https://github.com/GaloyMoney/galoy/pull/2082/commits/8f82da142c46affb0a0b024288326107af4d36ec

File: src/services/twilio.ts (hunk 1 above)
Comment (vindard) [reply src/services/twilio.ts]: Done: https://github.com/GaloyMoney/galoy/pull/2082/commits/8f82da142c46affb0a0b024288326107af4d36ec

**Review by dolcalmi — APPROVED** [review]

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**